cmd/necoperf-cli/cmd/completion.go

```diff
@@ -0,0 +1,105 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	k8sConfig "sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+func podCandidates(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	if len(args) != 0 {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	cfg, err := k8sConfig.GetConfig()
+	if err != nil {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+	k8sClient, err := client.New(cfg, client.Options{})
+	if err != nil {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	pods := &corev1.PodList{}
+	err = k8sClient.List(context.Background(), pods, client.InNamespace(config.namespace))
+	if err != nil {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	var podNames []string
+	for _, pod := range pods.Items {
+		if !strings.HasPrefix(pod.Name, toComplete) {
+			continue
+		}
+		podNames = append(podNames, pod.Name)
+	}
+
+	return podNames, cobra.ShellCompDirectiveNoFileComp
+}
+
+func containerCompletionFunc(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	if len(args) == 0 {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	cfg, err := k8sConfig.GetConfig()
+	if err != nil {
+		return nil, cobra.ShellCompDirectiveError
+	}
+	k8sClient, err := client.New(cfg, client.Options{})
+	if err != nil {
+		return nil, cobra.ShellCompDirectiveError
+	}
+
+	pod := &corev1.Pod{}
+	err = k8sClient.Get(context.Background(), client.ObjectKey{
+		Namespace: config.namespace,
+		Name:      args[0],
+	}, pod)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Is this intentional?
I'm afraid you forget to remove a debug message.

Though it would be nice to show an error message to a user when a runtime error occurs, it seems somewhat unnatural to show only for `k8sClient.Get()`.
To show a message, `cobra.AppendActiveHelp()` might be more appropriate than `fmt.Println()`.

Reply:
@morimoto-cybozu 
Sorry, I forgot to delete the debug message.

I will try to show the error message when an Error occurs at runtime.
I agree that using `cobra.AppendActiveHelp()` is better than `fmt.Println()`, but I felt that using `cobra.CompErrorln()` was better.
What do you think?

Reply:
@zeroalphat 
Great!
I've missed that function.

Reply:
@morimoto-cybozu 
Thank you very much.
I have changed this function to show the error